src/framework/components/rigid-body/system.js

```diff
@@ -552,6 +560,240 @@ class RigidBodyComponentSystem extends ComponentSystem {
         return results;
     }
 
+    /**
+     * Perform a collision check on the world and return all entities the shape hits.
+     * It returns an array of {@link RaycastResult}, one for each hit. If no hits are
+     * detected, the returned array will be of length 0.
+     *
+     * @param {object} shape - The shape to use for collision.
+     * @param {number} shape.axis - The local space axis with which the capsule, cylinder or cone shape's length is aligned. 0 for X, 1 for Y and 2 for Z. Defaults to 1 (Y-axis).
+     * @param {Vec3} shape.halfExtents - The half-extents of the box in the x, y and z axes.
+     * @param {number} shape.height - The total height of the capsule, cylinder or cone from tip to tip.
+     * @param {string} shape.type - The type of shape to use. Available options are "box", "capsule", "cone", "cylinder" or "sphere". Defaults to "box".
+     * @param {number} shape.radius - The radius of the sphere, capsule, cylinder or cone.
+     * @param {Vec3} position - The world space position for the shape to be.
+     * @param {Vec3} rotation - The world space rotation for the shape to have.
+     *
+     * @returns {RaycastResult[]} An array of shapecast hit results (0 length if there were no hits).
```

Comment:
It's not clear to me what it means to raycast with a shape that has volume. For a ray, the point is the point of ray penetration and the normal is the normal on the surface at that point. What does it mean for a shape? And is the `Raycast` naming a bit confusing?

Reply:
When using `dynamicsWorld.contactTest` the callback gives only 1 point per collision. But that's true that it may confuse users. Regarding `RaycastResult` being returned it's because it has the same structure. Another solution could be to directly return an array of entities?

Reply:
Alright so yeah I just removed the RaycastResult and now it returns an array of entities instead. Does that sound better to you @willeastcott?

Reply:
I'm not sure, @MushAsterion. It sort of depends on the meaning of the point + normal that was returned in the `RaycastResult` that was returned before. If Ammo is returning useful information there, we should pass it back to the application. Can you maybe look into that?

Reply:
> If Ammo is returning useful information there, we should pass it back to the application. Can you maybe look into that?

The information is as crucial as it gets here, otherwise there is no way to tell where the collision occurred.

I agree `RaycastResult` is a little bit too specific, so either it could be renamed into `CastResult` (`CastingResult`?) or simply `class CastResult extends RaycastResult {}` e.g.

The last thing I want is just an array of entities without any kind of collision information.

Reply:
I'd probably go with `CastResult` or `HitResult`.

Reply:
> I'd probably go with `CastResult` or `HitResult`

Nice, I like the sound of `HitResult` and it's even shorter :+1:

Reply:
That's added!

Regarding point + normal here are the two definitions I used:
- `HitResult#point` The point on the entity where the hit occurred, in world space. When returned by a shapecast it is the first point found to collide, it does not have to be the closest.
- `HitResult#normal` The normal vector of the hit on the entity, in world space.